packages/itwin/tree-widget/src/components/trees/models-tree/ModelsTreeButtons.tsx

```diff
@@ -32,8 +32,25 @@ export interface ModelsTreeHeaderButtonProps extends TreeHeaderButtonProps {
   models: ModelInfo[];
 }
 
-/** @internal */
-export function useAvailableModels(imodel: IModelConnection) {
+/**
+ * Custom hook that creates props required to render `ModelsTreeComponent` header button.
+ * @public
+ */
+export function useModelsTreeButtonProps({
```

Comment:
Need a plan for how this function will change (in a non-breaking way) to fix https://github.com/iTwin/viewer-components-react/issues/999.

Reply:
I think `useModelsTreeButtonProps` can be updated to return callback that will be invoked with filtered paths something like `onTreeFilter` and it could return filtered out `models` array that is passes to `showAll/hideAll` buttons.